### ISSUE TITLE
Plugin throws assertion error on project open in Idea 2020.1

### ISSUE THREAD

**Issue by mdindoffer** [microsoft/azure-devops-intellij#329]
Every time I open a TFVC-enabled project, the plugin throws an exception:
<details>
  <summary>Exception stacktrace</summary>
  
  ```
java.lang.Throwable: Assertion failed
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:182)
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:191)
	at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService(ComponentManagerImpl.kt:481)
	at com.intellij.serviceContainer.ComponentManagerImpl.getLightService$intellij_platform_serviceContainer(ComponentManagerImpl.kt:443)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:386)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:379)
	at com.intellij.notification.EventLog.getProjectService(EventLog.java:579)
	at com.intellij.notification.EventLog.getLogModel(EventLog.java:83)
	at com.intellij.notification.EventLog.getStatusMessage(EventLog.java:111)
	at com.intellij.openapi.wm.impl.status.StatusPanel.updateText(StatusPanel.java:180)
	at com.intellij.openapi.wm.impl.status.InfoAndProgressPanel.setText(InfoAndProgressPanel.java:515)
	at com.intellij.openapi.wm.impl.status.IdeStatusBarImpl.lambda$setInfo$5(IdeStatusBarImpl.java:397)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:2319)
	at com.intellij.openapi.wm.impl.status.IdeStatusBarImpl.setInfo(IdeStatusBarImpl.java:395)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeListener(MessageBusImpl.java:549)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:143)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:472)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:432)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:420)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:404)
	at com.intellij.util.messages.impl.MessageBusImpl.lambda$createTopicHandler$3(MessageBusImpl.java:243)
	at com.sun.proxy.$Proxy142.setInfo(Unknown Source)
	at com.intellij.openapi.wm.StatusBar$Info.set(StatusBar.java:52)
	at com.intellij.openapi.wm.StatusBar$Info.set(StatusBar.java:35)
	at com.intellij.ui.components.labels.LinkLabel.setStatusBarText(LinkLabel.java:284)
	at com.intellij.ui.components.labels.LinkLabel.disableUnderline(LinkLabel.java:276)
	at com.intellij.ui.components.labels.LinkLabel.removeNotify(LinkLabel.java:202)
	at java.desktop/java.awt.Container.removeNotify(Container.java:2831)
	at java.desktop/javax.swing.JComponent.removeNotify(JComponent.java:4809)
	at java.desktop/java.awt.Container.removeNotify(Container.java:2831)
	at java.desktop/javax.swing.JComponent.removeNotify(JComponent.java:4809)
	at java.desktop/java.awt.Container.removeAll(Container.java:1322)
	at com.intellij.openapi.wm.impl.status.InfoAndProgressPanel.removeAll(InfoAndProgressPanel.java:393)
	at com.intellij.openapi.wm.impl.status.InfoAndProgressPanel.buildInProcessCount(InfoAndProgressPanel.java:343)
	at com.intellij.openapi.wm.impl.status.InfoAndProgressPanel.addProgress(InfoAndProgressPanel.java:237)
	at com.intellij.openapi.wm.impl.status.IdeStatusBarImpl.addProgress(IdeStatusBarImpl.java:410)
	at com.intellij.openapi.progress.impl.BackgroundableProcessIndicator.doBackground(BackgroundableProcessIndicator.java:115)
	at com.intellij.openapi.progress.impl.BackgroundableProcessIndicator.<init>(BackgroundableProcessIndicator.java:54)
	at com.intellij.openapi.progress.impl.BackgroundableProcessIndicator.<init>(BackgroundableProcessIndicator.java:40)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.lambda$runProcessWithProgressAsynchronously$2(ProgressManagerImpl.java:107)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at com.intellij.util.concurrency.PlainEdtExecutor.execute(PlainEdtExecutor.java:15)
	at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1714)
	at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:1931)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressAsynchronously(ProgressManagerImpl.java:106)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runAsynchronously(CoreProgressManager.java:349)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:336)
	at com.intellij.openapi.progress.Task.queue(Task.java:109)
	at com.microsoft.alm.plugin.idea.tfvc.core.TFSVcs.lambda$checkCommandLineVersion$1(TFSVcs.java:341)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:310)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:80)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:128)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:184)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:847)
	at com.intellij.ide.IdeEventQueue.lambda$null$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:739)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
  ```
  
</details>

```
Azure DevOps plugin version: 1.159.0
IntelliJ IDEA 2020.1 (Ultimate Edition)
Build #IU-201.6668.121, built on April 8, 2020
Licensed to X
Subscription is active until August 8, 2020
Runtime version: 11.0.6+8-b765.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.0-46-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1954M
Cores: 4
Non-Bundled Plugins: Error-prone plugin, com.microsoft.vso.idea, com.urswolfer.intellij.plugin.gerrit, com.jetbrains.fast.mouse.scroll, com.mnr.java.intellij.idea.plugin.base64helper, MavenRunHelper, some.awesome
Current Desktop: X-Cinnamon
```

**Comment by ForNeVeR** [comment]
Thanks for your report. Unfortunately, I wasn't able to reproduce that. How stable is it? Which exact actions do you perform to get this issue?

Also, it seems it isn't an actual error, but just an assertion inside of IDEA code. It _should_ be harmless.

**Comment by ForNeVeR** [comment]
It looks like this issue is somehow linked to unloadable plugins. Is it possible that some of the plugins were dynamically loaded/unloaded in the session when you've encountered this error? Does it still reproduce?

**Comment by mdindoffer** [comment]
> Is it possible that some of the plugins were dynamically loaded/unloaded in the session when you've encountered this error?

How do I check for that?
Is there a way a plugin could load/unload without user interaction? If there isn't, then probably no, since I don't fiddle with the plugins (enable/disable them).

However, these days the assertion error does not get thrown every time on project open, contrary to my original post. The chance is about 60%? I found the most probable way to trigger this is during Intellij startup (exit the IDE and then start it again).

I tried to dissect the logs, but did not see anything of interest. I can provide the logs, if you want, but would have to anonymize them first.

**Comment by ForNeVeR** [comment]
Well, okay, let's wait for IDEA 2020.1.1 to come out. I still think that this assertion message isn't a very harmful one, and IDEA Core team is working on a fix for a similar issue. It may be linked with #330, but I'm not very sure about it. _Probably_, fix for #330 would fix this issue, too. We'll see.

(I still wasn't able to reproduce it even after #330.)

**Comment by ForNeVeR** [comment]
IDEA 2020.1.1 is out, so please reopen if you'll be able to reproduce this issue in it.
